sympy/solvers/solvers.py

```diff
@@ -694,6 +734,10 @@ def _sympified_list(w):
         elif isinstance(fi, (bool, C.BooleanAtom)) or fi.is_Relational:
             return reduce_inequalities(f, symbols=symbols)
```

Comment:
Should we check the solution from `reduce_inequality` before returning it, like we are doing for other cases? Specifically, should we use `_simple_dens` or `denoms` to check if the denominators are zero for any value in the interval or solution and exclude that value from the interval? See [#8715] (https://github.com/sympy/sympy/issues/8715).